pymchelper/averaging.py

```diff
@@ -0,0 +1,214 @@
+"""
+Output from the simulation running on multiple processess needs to be agregated.
+The simplest way of doing so is to calculated the average of the data.
+There are however more sophisticated cases: COUNT scorers needs to be summed up, not averaged.
+Phase space data needs to be concatenated, not averaged.
+Each of the parallel jobs can have different number of histories, so the weighted average needs to be calculated.
+Moreover, in case averaging is employed, we can estimate spread of the data as standard deviation or standard error.
+
+The binary output files from each job may be quite large (even ~GB for scoring in fine 3D mesh), to obtain good
+statistics we sometimes parallelise the simulation of hundreds or thousands of jobs (when using HPC clusters).
+In such case it's not feasible to load all the data into memory and calculate the average in one go, using standard
+functions from numpy library. Instead, we need to calculate the average in an online manner,
+i.e. by updating the state of respective aggregator object with each new binary output file read.
+
+Such approach results in a significant reduction of memory usage and is more numerically stable.
+
+This module contains several classes for aggregating data from multiple files:
+- Aggregator: base class for all other aggregators
+- WeightedStatsAggregator: for calculating weighted (using reliability, not frequency weights) mean and variance
```

Comment:
what is reliability?

Reply:
Jargon from statistics, I wasn't familiar with them before as well.

Reliability weights are the weights which do not sum up to the 1. For example w_1=100, w_2 = 3000, w_3 = 2000. This is in contrary to frequency weights like w_1 = 0.1, w_2 = 0.4, w_3 = 0.5.

See for example:
https://en.wikipedia.org/wiki/Weighted_arithmetic_mean

![image](https://github.com/DataMedSci/pymchelper/assets/7374733/6717cf0e-6986-46ca-9f24-58984600a547)